### PR TITLE
Taskbundles

### PR DESCRIPTION
**Pull request by gardnervickers**

### CODE REVIEW

**Comment by gardnervickers** [comment]
Good to go @lbradstreet @MichaelDrogalis. Also we can close #14 now too.

File: circle.yml [hunk 1]
```diff
@@ -6,14 +6,20 @@ dependencies:
...
-        timeout: 480 
+    - lein with-profile dev,circle-ci test :all:
+        timeout: 480
 
 machine:
   java:
     version: oraclejdk8
   services:
     - docker
+  pre:
```
Comment (MichaelDrogalis) [inline circle.yml]: Remove commented block.

File: src/onyx/tasks/datomic.clj [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+              (s/optional-key :onyx/n-peers) (s/enum 1)
+              UserTaskMapKey s/Any}]))
+
+(s/defn ^:always-validate read-log
+  ([task-name :- s/Keyword opts]
+   {:task {:task-map (merge {:onyx/name task-name
+                             :onyx/plugin :onyx.plugin.datomic/read-log
+                             :onyx/type :input
+                             :onyx/medium :datomic
+                             }
```
Comment (MichaelDrogalis) [inline src/onyx/tasks/datomic.clj]: Cosmetic: move up a line

File: test/onyx/plugin/input_fault_tolerance_test.clj [hunk 3]
```diff
@@ -1,34 +1,66 @@
...
+(defn build-job [db-uri t batch-size batch-timeout]
+  (let [batch-settings {:onyx/batch-size batch-size :onyx/batch-timeout batch-timeout}
+        base-job (merge {:workflow [[:read-datoms :query]
+                                    [:query :persist]]
+                         :catalog [{:onyx/name :query
+                                    :onyx/fn ::my-test-query
+                                    :onyx/type :function
+                                    :onyx/batch-size batch-size
+                                    :onyx/doc "Queries for names of 5 characters or fewer"}]
+                         :lifecycles [;{:lifecycle/task :read-datoms
```
Comment (MichaelDrogalis) [inline test/onyx/plugin/input_fault_tolerance_test.clj]: Remove dead code

File: test/onyx/plugin/input_log_test.clj [hunk 4]
```diff
@@ -173,68 +84,62 @@
...
+      (with-test-env [test-env [4 env-config peer-config]]
+        (testing "That we can read the initial transaction log"
+          (let [job (build-job db-uri 1002 10 1000)
+                {:keys [persist]} (get-core-async-channels job)
+                job-id (atom nil)]
+            (mapv (partial ensure-datomic! db-uri) [schema people])
+            (reset! job-id (:job-id (onyx.api/submit-job peer-config job)))
+            (ensure-datomic! db-uri people2)
+            (onyx.api/await-job-completion peer-config @job-id)
+            (is (= [{:data '(;[13194139534312 50 #inst "2015-08-19T13:27:59.237-00:00" 13194139534312 true]
```
Comment (MichaelDrogalis) [inline test/onyx/plugin/input_log_test.clj]: Drop commented code

File: test/onyx/plugin/input_log_test.clj [hunk 5]
```diff
@@ -173,68 +84,62 @@
...
+                             [63 10 :com.mdrogalis/people 13194139534312 true]
+                             [0 11 63 13194139534312 true]
+                             [64 10 :user/name 13194139534312 true]
+                             [64 40 23 13194139534312 true]
+                             [64 41 35 13194139534312 true]
+                             [0 13 64 13194139534312 true]) :t 1000}
+                    {:data '(;[13194139534313 50 #inst "2015-08-19T13:27:59.256-00:00" 13194139534313 true]
```
Comment (MichaelDrogalis) [inline test/onyx/plugin/input_log_test.clj]: Drop commented code

**Comment by MichaelDrogalis** [comment]
@gardnervickers Just a few cosmetic things. Fix those then merge it. :+1:

**Pull request merged**